### DISCUSSION TITLE
调整 linux 平台的 pyinstaller 自动构建

### DISCUSSION THREAD

**Discussion by pizeroLOL** [Class-Widgets/Class-Widgets#194] [调查]
1. pyinstaller 构建出的二进制文件的 Glibc 版本需要小于用户机器上的
2. Issue 模板里的提示会让用户使用 pyinstaller 打包后的内容，进而引发问题 1，如 https://github.com/Class-Widgets/Class-Widgets/issues/192
3. Glibc 问题可以通过降低容器版本缓解，但问题是现在已经降到最低了，再降就要引入其他工具，如 docker；或用后处理工具，如 staticx 解决

**Comment by MKStoler4096** [comment]
主要是有以下几点原因：

1. 用 Actions 打包的用户大多都不会 Linux 。（我甚至不会装 Python ）
2. 大多数用户都在用的是统信UOS。

**Reply by pizeroLOL** [reply]
1. 待议
2. 有数据吗，比如多少人之类的

**Reply by MKStoler4096** [reply]
> 1. 待议
> 2. 有数据吗，比如多少人之类的

新希沃全是 UOS ，可以自己查一下

**Comment by pizeroLOL** [comment]
docker

- python 3.8-buster Glibc 2.28
- python 3.7-stretch Glibc 2.24 涉及到降级，得改依赖，pyqt-sip 不是很好装

staticx 在 ubuntu 20.04 docker 镜像里搞了一下午都没能在 archlinux 里跑起来
